autoemulate/experimental/save.py

```diff
@@ -0,0 +1,169 @@
+import ast
+from pathlib import Path
+
+import joblib
+import pandas as pd
+
+from autoemulate.experimental.emulators.base import Emulator
+from autoemulate.experimental.results import Result  # , Results
+
+
+class ModelSerialiser:
+    def __init__(self, logger):
+        self.logger = logger
+
+    def _save_model(
+        self,
+        model: Emulator,
+        model_name: str | None = None,
+        path: str | Path | None = None,
+    ):
+        """Saves a model to disk.
+
+        Parameters
+        ----------
+        model : Emulator model
+            Model to save.
+        model_name : str or None, optional
+            Name of the model to save.
+            If None, the model's name will be used.
+        path : str, Path or None, optional
+            Path to save the model.
+            If None, the model will be saved with the model name.
+        """
+        model_filename = self._get_model_filename(model, model_name)
+        full_path = self._prepare_path(path, model_filename)
+        try:
+            if full_path.suffix != ".joblib":
+                joblib.dump(model, full_path.with_suffix(".joblib"))
```

Comment:
It looks like this might replace any current suffix with joblib - I think this is causing the failing test as `.pt` is a suffix in the test?